packages/react-bootstrap-table2/src/props-resolver/index.js

```diff
@@ -83,11 +83,11 @@ export default ExtendBase =>
       if (_.isDefined(selectRow)) {
         let checkedStatus;
 
-        const allRowsSelected = data.length === selectedRowKeys.length;
+        const allRowsSelected = this.store.isAllRowsSelected();
```

Comment:
After discussion, I decided to handle all logic for row selection with in store and omit the situation for pagination. This PR will focus on row selection which **ignores pagination**.